lua/barbar/buffer.lua

```diff
@@ -149,9 +149,6 @@ local buffer = {
       else
         name = strcharpart(name, 0, maximum_length - ELLIPSIS_LEN) .. ELLIPSIS
       end
-
-      -- safety to prevent recursion in any future edge case
-      name = name:sub(1, maximum_length)
```

Comment:
I tried to make this recurse by doing `return name .. ' '` at the end, and it didn't blow the stack or anything. Seems safe to take it out, since we have the `slice` methods anyway in the render code that can clean up here if things mess up.

If you object to it, I can always use `strcharpart` instead, but that is ~6× slower than `string.sub` or just taking it out.

Reply:
Let's remove, we have no reason to be defensive here.